nessie/jobs/refresh_canvas_data_catalog.py

```diff
@@ -0,0 +1,141 @@
+"""
+Copyright ©2019. The Regents of the University of California (Regents). All Rights Reserved.
+
+Permission to use, copy, modify, and distribute this software and its documentation
+for educational, research, and not-for-profit purposes, without fee and without a
+signed licensing agreement, is hereby granted, provided that the above copyright
+notice, this paragraph and the following two paragraphs appear in all copies,
+modifications, and distributions.
+
+Contact The Office of Technology Licensing, UC Berkeley, 2150 Shattuck Avenue,
+Suite 510, Berkeley, CA 94720-1620, (510) 643-7201, otl@berkeley.edu,
+http://ipira.berkeley.edu/industry-info for commercial licensing opportunities.
+
+IN NO EVENT SHALL REGENTS BE LIABLE TO ANY PARTY FOR DIRECT, INDIRECT, SPECIAL,
+INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING LOST PROFITS, ARISING OUT OF
+THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN IF REGENTS HAS BEEN ADVISED
+OF THE POSSIBILITY OF SUCH DAMAGE.
+
+REGENTS SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
+SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF ANY, PROVIDED HEREUNDER IS PROVIDED
+"AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,
+ENHANCEMENTS, OR MODIFICATIONS.
+"""
+
+
+from datetime import datetime, timedelta
+
+from flask import current_app as app
+from nessie.externals import canvas_data, redshift
+from nessie.jobs.background_job import BackgroundJob, BackgroundJobError, verify_external_schema
+from nessie.lib.util import get_s3_canvas_daily_path
+import pandas as pd
+
+
+"""Logic for generate canvas data catalog job."""
+
+
+class RefreshCanvasDataCatalog(BackgroundJob):
+
+    def run(self):
+        # Retrieve latest schema definitions from Canvas data API
+        response = canvas_data.get_canvas_data_schema(mock=None)
```

Comment:
@sandeepmjay It should be safe to leave `mock=None` out of this line; that argument is only something we'd hook into during a test.